public/API/API_GetUserProfile.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ *  API_GetUserProfile
+ *    u : username
+ *
+ *  string     User                    name of user
+ *  int        ID                      unique identifier of the user
+ *  int        TotalPoints             number of hardcore points the user has
+ *  int        TotalSoftcorePoints     number of softcore points the user has
+ *  int        TotalTruePoints         number of "white" points the user has
+ *  int        Permissions             unique identifier of user's account type
```

Comment:
Is there some benefit to exposing the user's permissions? We're going to be overhauling that functionality in the near future.

Reply:
Same as `UserWallActive` - shouldn't be included but it's present in summary, too.

Do we want to remove it from both?

Reply:
`Permissions` seems reasonable to remove.

It may make sense for a future iteration of the API allows someone to read a user's wall posts. That's the only reason I can think of to continue exposing `UserWallActive`.

Reply:
API_GetUserSummary just dumps the data returned by the query for populating the user page, which does display the user's "Account Type" (permissions) and the user wall.

Reply:
We currently use the Permissions field to check if a user is banned from the site and remove them from tracking. I don't know if it would be better to just a return a 404 for these types of users (as looking on the site itself, a not found page is returned when accessing a banned profile it seems). 

But I just kept the behavior the same as API_GetUserSummary in that regard - maybe it's better to tackle that in a future iteration of the API.

Reply:
I agree, let's keep it as is for now?
I'd consider this iteration of the API to be v1 and we should not change it too much to meet current clients' expectations.
Using dedicated resource classes which map to the current schema and payload would make it more robust - this would prevent unexpected payload changes because of query changes. We now have characterization tests for this in place to refactor such things confidently. I'll look into that in a separate PR.